### PR TITLE
Query Splitting: Fix for handling queries with no requestId

### PR DESCRIPTION
**Pull request by domasx2**
Hey!

Found that when `lokiQuerySplitting` is enabled, all queries except the last one issued by Faro plugin get cancelled. Tracked it down to[ this change](https://github.com/grafana/grafana/pull/63663). The root cause is that query splitting mechanism assigns request ids for split queries that are `${originalRequestId}_${splitRequestIndex}`. However, request may not have a `requestId`, like in case of Faro, and end up being assigned `requestId='undefined_1'`. Then multiple concurrent but unrelated queries end up having the same `requestId`, and all but the last one get cancelled by `backendSrv`.

This PR fixes it by not assigning a `requestId` to split query request if it didn't exist on the original request.

### CODE REVIEW

File: public/app/plugins/datasource/loki/querySplitting.ts [hunk 1]
```diff
@@ -120,23 +119,27 @@ export function runSplitGroupedQueries(datasource: LokiDatasource, requests: Lok
...
-          nextRequest();
-        },
-        error: (error) => {
-          subscriber.error(error);
-        },
-      });
+    const subRequest = { ...requests[requestGroup].request, range, targets };
+    // request may not have a request id
+    if (group.request.requestId) {
+      subRequest.requestId = `${group.request.requestId}_${requestN}`;
```
Comment (matyax) [inline public/app/plugins/datasource/loki/querySplitting.ts:125]: I'm not really sure of the purpose of the `requestId`, but I wonder if we shouldn't always define one, like

```javascript
      subRequest.requestId = subRequest.requestId ? `${group.request.requestId}_${requestN}` : `${requestN}`;
```

**Review by matyax — APPROVED** [review]
Great find! Thank you for implementing the fix 🙌

File: public/app/plugins/datasource/loki/querySplitting.ts (hunk 1 above)
Comment (domasx2) [reply public/app/plugins/datasource/loki/querySplitting.ts:125]: That would still end up assigning identical request IDs , '1',  to unrelated requests and causing the problem.
Instead we could generate a random `requestId` [here](https://github.com/grafana/grafana/pull/66161/files/3b654bdd3ba5c75aa21d1fffec7f45ca4af577af#diff-623c3cc7d36f876a6b5994d057bc91dbad65060046cddfe6fc68df9de9ca8837R181) if one does not exist. 
But I think if one was not assigned to begin with, it's not really needed at all

Comment (matyax) [reply public/app/plugins/datasource/loki/querySplitting.ts:125]: You're right.

> But I think if one was not assigned to begin with, it's not needed at all

Yes, perfect. I just wanted to double check if we should always have one or not. Thanks!

**Pull request merged**